### ISSUE TITLE
Extend sonata_type_model_autocomplete

### ISSUE THREAD

**Issue by MM-Psiiirus** [sonata-project/SonataAdminBundle#3164]
Hiya,
i'm trying to extend the current sonata_type_model_autocomplete field to add some more logic to it but unfortunately the ajax callback seems to be hardcoded to the "sonata_type_model_autocomplete" type:
```
in vendor/sonata-project/admin-bundle/Controller/HelperController.php at line 453 

        if ($fieldDescription->getType() !== 'sonata_type_model_autocomplete') {
            throw new \RuntimeException(sprintf('Unsupported form type "%s" for field "%s".', $fieldDescription->getType(), $field));
```

Is there any configurable way to get around this issue or do i have to hack the ajax call get it work with another form_field?

Cheers,
Toni

**Comment by maurermax** [comment]
Okay, I just had the same problem and the most non invasive fix that I found was redirecting the `/core/get-autocomplete-items` route to a controller that I created myself extending the HelperController and just overriding the `retrieveAutocompleteItemsAction` `retrieveFieldDescription` and creating the `$pool` variable locally as this accessed without a getter. 

Then I removed the lines checking this and everything now works fine. Contact me if you need more infos on this.

**Comment by ElectricMaxxx** [comment]
Maybe @rande can give us a hand with that, cause https://github.com/sonata-project/SonataAdminBundle/issues/3341 is related to that and cause by re-implementing stuff like that: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/493

**Comment by pulzarraider** [comment]
Fixed with #3407. @MM-Psiiirus @maurermax thank you for reporting the issue.
